signup/index.js

```diff
@@ -0,0 +1,87 @@
+const INPUT_EMAIL = document.querySelector('#signup-email');
+const INPUT_PASSWORD = document.querySelector('#signup-password');
+const INPUT_PASSWORD_CHECK = document.querySelector('#check-password')
+const ERROR_MESSAGE = document.querySelector('.error_message');
+const FORM_ELEMENT = document.querySelector('#signup-form');
+
+INPUT_EMAIL.addEventListener('focusout', handleEmailFocusout);
+INPUT_PASSWORD.addEventListener('focusout', handlePasswordFocusout);
+INPUT_PASSWORD_CHECK.addEventListener('focusout', handlePasswordCheckFocusout);
+FORM_ELEMENT.addEventListener('submit', handleSubmit);
+
+
+// 이메일 유효성 확인
+function emailValidation(input) {
+  const REG_EMAIL = /^([0-9a-zA-Z_\.-]+)@([0-9a-zA-Z_-]+)(\.[0-9a-zA-Z_-]+){1,2}$/;
+  return REG_EMAIL.test(input);
+}
+
+function handleEmailFocusout(e) {
+  if (INPUT_EMAIL.value === '') {
+    INPUT_EMAIL.nextElementSibling.textContent = '이메일을 입력해 주세요.';
+    INPUT_EMAIL.classList.add('input_error');
+
+    return false;
+
+  } else if (INPUT_EMAIL.value !== '') {
+    if (INPUT_EMAIL.value === 'test@codeit.com') {
+      INPUT_EMAIL.nextElementSibling.textContent = '이미 사용 중인 이메일입니다.';
+      INPUT_EMAIL.classList.add('input_error');
+      return false;
+    } else if (!emailValidation(INPUT_EMAIL.value)) {
+      INPUT_EMAIL.nextElementSibling.textContent = '올바른 이메일 주소가 아닙니다.';
+      INPUT_EMAIL.classList.add('input_error');
+      return false;
+    }
+  }
+}
```

Comment:
if~else 내에서 블록이 종료되기 전 `return false`코드가 있기 때문에 이후 로직들은 실행되지 않습니다.
그래서 아래처럼 코드를 수정할 수 있을 것 같습니다.
```suggestion
function handleEmailFocusout(e) {
  if (INPUT_EMAIL.value === '') {
    INPUT_EMAIL.nextElementSibling.textContent = '이메일을 입력해 주세요.';
    INPUT_EMAIL.classList.add('input_error');

    return false;

  }
  
  if (INPUT_EMAIL.value === 'test@codeit.com') {
    INPUT_EMAIL.nextElementSibling.textContent = '이미 사용 중인 이메일입니다.';
    INPUT_EMAIL.classList.add('input_error');
    return false;
  }
  
  if (!emailValidation(INPUT_EMAIL.value)) {
    INPUT_EMAIL.nextElementSibling.textContent = '올바른 이메일 주소가 아닙니다.';
    INPUT_EMAIL.classList.add('input_error');
    return false;
  }
  
  return true;
}

```